src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs

```diff
@@ -24,26 +29,155 @@ private DesigntimeLicenseContextSerializer()
         /// using the specified key and output stream.
         /// </summary>
         public static void Serialize(Stream o, string cryptoKey, DesigntimeLicenseContext context)
+        {
+            if (EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization)
+            {
+                SerializeWithBinaryFormatter(o, cryptoKey, context);
+            }
+            else
+            {
+                using (BinaryWriter writer = new BinaryWriter(o, encoding: Text.Encoding.UTF8, leaveOpen: true))
+                {
+                    writer.Write(BinaryWriterMagic); // flag to identify BinaryWriter
+                    writer.Write(cryptoKey);
+                    writer.Write(context._savedLicenseKeys.Count);
+                    foreach (DictionaryEntry keyAndValue in context._savedLicenseKeys)
+                    {
+                        writer.Write(keyAndValue.Key.ToString());
+                        writer.Write(keyAndValue.Value.ToString());
+                    }
+                }
+            }
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
```

Comment:
All the usages of UnconditionalSuppressMessage I saw used "ReflectionAnalysis" and "IL2026:UnrecognizedReflectionPattern" as the parameters to the attribute. I think it fits here too, but calling it out in case I need to use something else. @eerhardt

Reply:
```suggestion
        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
```

The existing places that have `IL2026:UnrecognizedReflectionPattern` are a mistake.